Data/Scripts/GardenConquest/Blocks/GridEnforcer.cs

```diff
@@ -416,11 +419,20 @@ public enum VIOLATION_TYPE {
 		/// </summary>
 		/// <param name="b"></param>
 		private void removeBlock(IMySlimBlock b) {
-			// TODO: spawn materials in place so they're not lost
-			// Need to find a way to get the stockpile status
-			// Currently locked behind MySlimBlock (non-interface)
+			// spawn materials in place so they're not lost
+
+			// TODO: determine best inventory target
+			IMyInventory inventory = null;
+
+			// WAITING: once Keen accepts this PR
+			// https://github.com/KeenSoftwareHouse/SpaceEngineers/pull/52
+			// the below will spawn materials in the inventory if it exists and has space,
+			// or otherwise as floating objects in space,
+			//b.FullyDismount(inventory);
+			//b.MoveItemsFromConstructionStockpile(inventory);
+			//b.SpawnConstructionStockpile();
 
-			m_Grid.RemoveBlock(b);
+			b.CubeGrid.RemoveBlock(b);
```

Comment:
Previous line and your change are identical.  If a grid splits a new one is created and new block objects are allocated.  b.CubeGrid will still be m_Grid even if it is actually no longer attached.